app/single_resource/views.py

```diff
@@ -171,11 +171,23 @@ def save_associations(resource, form, descriptors, resource_existed=True):
         for value in values:
             association = None
             if resource_existed:
-                association = AssociationClass.query.filter_by(
+                associations = AssociationClass.query.filter_by(
                     resource_id=resource.id,
                     descriptor_id=descriptor.id
-                ).first()
+                )
+                for a in associations:
+                    print a
+                    db.session.delete(a)
+                    try:
+                        db.session.commit()
+                        flash('Resource deleted', 'form-success')
```

Comment:
Actually to clarify, that was my attempt at fixing the editing bug. It has now been commented out because if effectively does nothing.